### ISSUE TITLE
`validate=False` seemingly has no effect in `to_dict()` for encoding channel classes

### ISSUE THREAD

**Issue by joelostblom** [altair-viz/altair#3075]
While `alt.Chart().to_dict(validate=False)` bypasses validation as expected, it seems like the same is not true for the encoding classes. For example, `alt.Color('test').to_dict(validate=False)` raises `ValueError: test encoding field is specified without a type`

**Comment by binste** [comment]
I agree that it's more intuitive if we don't raise the error if `validate=False` and so that it's a bug. One minor addition: `validate=False` still disables the jsonschema validation for channels but we don't even get that far as there is this ValueError which is raised already in `FieldChannelMixin.to_dict`.

**Comment by dangotbanned** [comment]
@joelostblom what would you like this to return?

```py
import alt
alt.Color('test').to_dict(validate=False)
```

This would be one option:

```diff
diff --git a/altair/vegalite/v5/schema/channels.py b/altair/vegalite/v5/schema/channels.py
index a4a676c23..d3d6c723f 100644
--- a/altair/vegalite/v5/schema/channels.py
+++ b/altair/vegalite/v5/schema/channels.py
@@ -207,6 +207,8 @@ class FieldChannelMixin:
                         " also confirm that the data type is specified correctly."
                     )
                     raise ValueError(msg)
+                elif not validate:
+                    parsed = {"field": shorthand}
                 else:
                     msg = (
                         f"{shorthand} encoding field is specified without a type; "
```
Which outputs:

```py
{'field': 'test'}
```

The other option, which I can't see a way towards is:
```py
{'shorthand': 'test'}
```

---

What would be the use-case for this btw?
Does it outweigh the risk of breakage, when trying to rewrite all of this pretty fragile code?

https://github.com/vega/altair/blob/b6253ffc120c5a03661a6ec55e3c9375d716d873/tools/schemapi/schemapi.py#L1183-L1186

https://github.com/vega/altair/blob/b6253ffc120c5a03661a6ec55e3c9375d716d873/tools/schemapi/schemapi.py#L1381-L1404

https://github.com/vega/altair/blob/b6253ffc120c5a03661a6ec55e3c9375d716d873/altair/utils/core.py#L515-L699

**Comment by joelostblom** [comment]
> What would be the use-case for this btw? Does it outweigh the risk of breakage, when trying to rewrite all of this pretty fragile code?

Probably not. I don't remember the exact use case I had when opening this issue. I think i just tried it and reacted to that it worked differently from `Chart.to_dict(validate=False)`. Maybe we can close for now and if it comes up for someone again we can revisit.
